binaries/speed_benchmark.cc

```diff
@@ -99,23 +102,44 @@ int main(int argc, char** argv) {
         CAFFE_ENFORCE(caffe2::ReadProtoFromFile(input_files[i], &blob_proto));
         workspace->CreateBlob(input_names[i])->Deserialize(blob_proto);
       }
-    } else if (caffe2::FLAGS_input_dims.size()) {
+    } else if (caffe2::FLAGS_input_dims.size() || caffe2::FLAGS_input_type.size()) {
+      CAFFE_ENFORCE_NE(0, caffe2::FLAGS_input_dims.size(),
+          "Input dims must be specified when input tensors are used.");
+      CAFFE_ENFORCE_NE(0, caffe2::FLAGS_input_type.size(),
```

Comment:
nit: consider allowing user to not specify types and default to `float `?

Reply:
Many models use `uint8_t` input with `PackedInt8BGRANHWCToNCHWCStylizerPreprocess` as the first operation, so I'd rather have users think about the data type explicitly.